spec/unit/actions/package_delete_spec.rb

```diff
@@ -24,18 +24,32 @@ module VCAP::CloudController
             TestConfig.override(packages: { image_registry: { base_path: 'hub.example.com/user' } })
           end
 
-          it 'schedules a job to the delete the blobstore item' do
-            expect {
-              package_delete.delete(package)
-            }.to change {
-              Delayed::Job.count
-            }.by(1)
+          context 'when the package type is bits' do
+            it 'schedules a job to the delete the package from the container registry' do
+              expect {
+                package_delete.delete(package)
+              }.to change {
+                Delayed::Job.count
+              }.by(1)
 
-            job = Delayed::Job.last
-            expect(job.handler).to include('VCAP::CloudController::Jobs::Kubernetes::RegistryDelete')
-            expect(job.handler).to include(package.bits_image_reference(digest: false))
-            expect(job.queue).to eq(Jobs::Queues.generic)
-            expect(job.guid).not_to be_nil
+              job = Delayed::Job.last
+              expect(job.handler).to include('VCAP::CloudController::Jobs::Kubernetes::RegistryDelete')
+              expect(job.handler).to include(package.bits_image_reference(digest: false))
+              expect(job.queue).to eq(Jobs::Queues.generic)
+              expect(job.guid).not_to be_nil
+            end
+          end
+
+          context 'when the package type is docker' do
```

Comment:
does the old task perform idempotent deletes, but the new one does not?

Reply:
My guess is this code was protecting it:
https://github.com/cloudfoundry/cloud_controller_ng/blob/68aa90a221e56d159586f7cd51449567ff2357c1/app/jobs/runtime/blobstore_delete.rb#L18-L19

I think the intent of it is to not try and delete something that was already deleted... but it probably shields docker packages as well.

Reply:
We were hoping to get this through by EOD for the cf-for-k8s 1.0 cutoff, so I was hesitant to change the behavior there.

Reply:
I wouldn't change the behavior, but a different way of framing the bug is to say that the RegistryDelete job is insufficiently idempotent

Reply:
> RegistryDelete job is insufficiently idempotent

The `RegistryDelete` job is idempotent. This error is actually being raised by the `package.bits_image_reference` method since there is no such thing as a `bits_image_reference` for a docker package.

If that method returned an empty string in this case we would then handle that in the `RegistryDelete` action and not call the registry-buddy... but ideally I think neither the `BlobstoreDelete` nor `RegistryDelete` should be called in the case of docker apps cause we _know_ there is no source code to delete.

Reply:
To summarize that a bit more succinctly...

`package.bits_image_reference` currently raises an error. We could have that method return `""` or `nil` in the docker package case.

The caveat is it kind of masks the fact that we don't really need to delete anything from a container registry for docker packages.

When @matt-royal and I were pairing on this code originally we valued the explicitness of the error raising... but now I have no strong feelings either way.

Reply:
> package.bits_image_reference currently raises an error

ahhh by the time I had dug this far I had lost sight of this